internal/olm/client/client.go

```diff
@@ -113,17 +113,44 @@ func (c Client) DoCreate(ctx context.Context, objs ...client.Object) error {
 	for _, obj := range objs {
 		kind := obj.GetObjectKind().GroupVersionKind().Kind
 		log.Infof("  Creating %s %q", kind, getName(obj.GetNamespace(), obj.GetName()))
-		err := c.KubeClient.Create(ctx, obj)
-		if err != nil {
-			if !apierrors.IsAlreadyExists(err) {
+
+		for i := 0; i < 5; i++ { // try to create 5 times before giving up
+			retry, err := c.tryCreatingOneResource(ctx, obj, kind)
+			if err != nil {
 				return err
 			}
-			log.Infof("    %s %q already exists", kind, getName(obj.GetNamespace(), obj.GetName()))
+
+			if retry {
+				log.Infof("    the CRD for %s %q is not ready yet. Retrying in one second", kind, getName(obj.GetNamespace(), obj.GetName()))
+				time.Sleep(time.Second)
+				continue
+			}
+
+			log.Infof("    %s %q created", kind, getName(obj.GetNamespace(), obj.GetName()))
+			break
 		}
 	}
 	return nil
 }
 
+func (c Client) tryCreatingOneResource(ctx context.Context, obj client.Object, kind string) (bool, error) {
+	err := c.KubeClient.Create(ctx, obj)
+	if err != nil {
+		if apierrors.IsAlreadyExists(err) {
+			log.Infof("    %s %q already exists", kind, getName(obj.GetNamespace(), obj.GetName()))
+			return false, nil
+		}
+
+		if meta.IsNoMatchError(err) {
```

Comment:
nit: could use an `if/else if/else` statement here to more clearly semantically signal that the cases being checked for are related.

Reply:
I guess it's a matter of style. I'm not sure that go-simple linter will agree here, for instance (`else` after `return`)